File: 백준/Silver/2559. 수열/수열.cc
#include <cstdio>
using namespace std;
int ar[100010];
int main()
{
int n, k;
scanf("%d %d", &n, &k);
for (int i = 1; i <= n; i++)scanf("%d", &ar[i]);
int Max = 0;
for (int i = 1; i <= k; i++) {
Max += ar[i];
}
int s = Max;
for (int i = 1; i <= n - k; i++) {
s = s - ar[i] + ar[i + k];
if (Max < s) {
Max = s;
}
}
printf("%d", Max);
return 0;
}

File: 백준/Silver/2559. 수열/수열.java
import java.util.*;
import java.io.*;

class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
		int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());
        
        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for(int i = 0; i<n; i++){
            arr[i] = Integer.parseInt(st.nextToken());
        }
        
        int answer =  0;
        for(int i = 0; i<k; i++){
            answer += arr[i];
        }
        
        int now = answer;
        for(int i = 1; i<=n-k; i++){
            now = now - arr[i-1] + arr[i+k-1];
            answer = Math.max(answer, now);
        }
        
        System.out.println(answer);
        
    }
}